x-pack/plugins/embeddable_enhanced/public/plugin.ts

```diff
@@ -49,16 +53,19 @@ export interface StartDependencies {
 // eslint-disable-next-line @typescript-eslint/no-empty-interface
 export interface SetupContract {}
 
+export interface ReactEmbeddableDynamicActionsApi {
+  dynamicActionsApi: HasDynamicActions;
+  dynamicActionsComparator: StateComparators<DynamicActionsSerializedState>;
+  serializeDynamicActions: () => DynamicActionsSerializedState;
+  startDynamicActions: () => { stopDynamicActions: () => void };
+}
```

Comment:
Originally, we were starting the `DynamicActionManager` in the `initializeDynamicActions` method - however, we previously didn't have a way to **stop** the manager on unmount - this resulted in drilldowns being duplicated as you navigated around dashboards. 

It felt cleaner to me that, rather than still starting in `initializeDynamicActions` and returning **just** the stop method in this API (in which case, the embeddable would only need to stop the manager on cleanup and wouldn't have to worry about starting the manager), we tie these two things together in the API - that way, the embeddable is responsible for **both** starting and stopping the manager.